src/Sylius/Behat/Context/Api/BaseContext.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api;
+
+use Behat\Behat\Context\Context;
+use Behat\Behat\Hook\Scope\BeforeStepScope;
+use Sylius\Behat\Service\SharedStorageInterface;
+
+abstract class BaseContext implements Context
+{
+    public function __construct(protected SharedStorageInterface $sharedStorage)
+    {
+    }
+
+    /** @BeforeStep */
+    public function beforeStep(BeforeStepScope $scope): void
+    {
+        $this->sharedStorage->set('step_name', $scope->getStep()->getText());
+    }
+
+    /** @AfterScenario */
+    public function afterScenario(): void
```

Comment:
```suggestion
    /** @BeforeScenario */
    public function beforeScenario(): void
```
After seems like an obvious choice, but if a scenario gets interrupted for whatever reason, the event won't get sent and we might be left with the previous error in the array. Basically, let's make it less stateful.